ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.cpp

```diff
@@ -42,6 +42,91 @@ inline bool use_multicore_device_tilize(
     return num_tiles_in_row <= max_tiles;
 }
 
+bool requires_padding_change(const ttnn::Tensor& tensor, ttnn::Layout layout) {
```

Comment:
what if tile size changes in tile layout?

Reply:
According to the comment:
```
// It's okay for conversion to tile layout to preserve arbitrary padding as long as it satisfies the alignment
```